worlds/yugioh06/utils.py

```diff
@@ -0,0 +1,27 @@
+# support for AP world
+import pathlib
+import sys
+import zipfile
+import io
+
+isAPWorld = ".apworld" in sys.modules[__name__].__file__
+
+
+def getZipFile():
+    filename = sys.modules[__name__].__file__
+    apworldExt = ".apworld"
+    zipPath = pathlib.Path(filename[:filename.index(apworldExt) + len(apworldExt)])
+    return (zipfile.ZipFile(zipPath), zipPath.stem)
```

Comment:
Now why is this necessary then, let me see what this does

Reply:
Ah, I see.

I recommend using pkgutil.get_data(), which is a generic file opening solution that should work with both zip and non zip.

You can find examples of this in many worlds across AP, it is the standard solution to the apworld zipfile problem.